transport/http/outbound.go

```diff
@@ -339,7 +349,27 @@ func (o *Outbound) getPeerForRequest(ctx context.Context, treq *transport.Reques
 
 func (o *Outbound) createRequest(treq *transport.Request) (*http.Request, error) {
 	newURL := *o.urlTemplate
-	return http.NewRequest("POST", newURL.String(), treq.Body)
+	hreq, err := http.NewRequest("POST", newURL.String(), treq.Body)
+	if err != nil {
+		return nil, err
+	}
+	// grpc in, http out case
+	if treq.Transport == "grpc" {
```

Comment:
requests reaching http outbound typically have transport as `http`, are we sure that `grpc` would be the transport name in grpc-in cases?